### ISSUE TITLE
Flux v2.2 requires an upgrade of HelmReleases before utilising new features

### ISSUE THREAD

**Issue by siegenthalerroger** [fluxcd/flux2#4479]
### Describe the bug

After updating to Flux 2.2 the usage of the previously experimental driftDetection must now be defined per-release. However when defining the `spec.driftDetection` an error is thrown by the kustomize controller pertaining to not being able to merge the old with the new spec. 

The following error is shown:
```
{"level":"error","ts":"2023-12-14T08:38:20.865Z","msg":"Reconciliation failed after 4.116645264s, next try in 5m0s","controller":"kustomization","controllerGroup":"kustomize.toolkit.fluxcd.io","controllerKind":"Kustomization","Kustomization":{"name":"foobar","namespace":"flux-system"},"namespace":"flux-system","name":"foobar","reconcileID":"a4510658-3d8d-4b9e-8694-61afc957ae82","revision":"main@sha1:ea1ea19bb5ebfe02d632ddd8a2468b9f003db5b4","error":"HelmRelease/gitlab/helm-gitlab-operator dry-run failed: failed to prune fields: failed to convert merged object to last applied version: .spec.driftDetection: field not declared in schema\n"}
```

To get it to work a manual, forced, helm upgrade "completes" the adoption of the old HelmRelease and allows the reconciliation to function with no further issue. 

### Steps to reproduce

1. Be on Flux v2.1
2. Have a HelmRelease defined with `v2beta1` version.
3. Update Flux to v2.2
4. Update your HelmRelease to apiVersion `v2beta2`
5. Flux reconciles this with no issue
6. Update your HelmRelease to include a `spec.driftDetection` section
7. Flux will now throw an error for the Kustomization

**If a manual Helm Upgrade is forced between steps 5 and 6, there is no error** (thankfully this is now possible with the `flux reconcile hr xxx --force` cli command).

### Expected behavior

This requirement should be documented or unnecessary. Flux should adopt the old HelmRelease and immediately be capable of merging the new spec together.

### OS / Distro

Ubuntu and/or Azure Linux

### Flux version

v2.2.0

### Flux check

```
► checking prerequisites
✔ Kubernetes 1.28.3 >=1.26.0-0
► checking version in cluster
✔ distribution: flux-v2.2.0
✔ bootstrapped: true
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.37.0
✔ image-automation-controller: deployment ready
► ghcr.io/fluxcd/image-automation-controller:v0.37.0
✔ image-reflector-controller: deployment ready
► ghcr.io/fluxcd/image-reflector-controller:v0.31.1
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v1.2.0
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v1.2.2
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v1.2.2
► checking crds
✔ alerts.notification.toolkit.fluxcd.io/v1beta3
✔ buckets.source.toolkit.fluxcd.io/v1beta2
✔ gitrepositories.source.toolkit.fluxcd.io/v1
✔ helmcharts.source.toolkit.fluxcd.io/v1beta2
✔ helmreleases.helm.toolkit.fluxcd.io/v2beta2
✔ helmrepositories.source.toolkit.fluxcd.io/v1beta2
✔ imagepolicies.image.toolkit.fluxcd.io/v1beta2
✔ imagerepositories.image.toolkit.fluxcd.io/v1beta2
✔ imageupdateautomations.image.toolkit.fluxcd.io/v1beta1
✔ kustomizations.kustomize.toolkit.fluxcd.io/v1
✔ ocirepositories.source.toolkit.fluxcd.io/v1beta2
✔ providers.notification.toolkit.fluxcd.io/v1beta3
✔ receivers.notification.toolkit.fluxcd.io/v1
✔ all checks passed
```

### Git provider

Gitlab

### Container Registry provider

ACR

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hiddeco** [comment]
We are going to investigate this, but I have high doubts that `flux reconcile helmrelease` would actually require a `--force` flag for it to start to work.

Simply requesting reconciliation of the resource without `--force` should have the same effect for any issue regarding the `apiVersion`.

**Comment by siegenthalerroger** [comment]
Thanks for the quick fix and the headsup that `--reset` should also be enough.

**Comment by hiddeco** [comment]
Should now also be solved by updating to `v2.2.1`: https://github.com/fluxcd/flux2/releases/tag/v2.2.1
